### PR TITLE
Update endian.h

### PR DESCRIPTION
**Pull request by DavidMarec**
`endian.h` has been introduced in FreeBSD13-Stable.

### CODE REVIEW

**Comment by tobhe** [comment]
Hopefully fixed with https://github.com/openiked/openiked-portable/commit/4316a760411cea20845b94ca9f0c36fb97f82bdc

**Comment by tobhe** [comment]
Indeed, can you rebase your fix to include that?

File: compat/endian.h [hunk 1]
```diff
@@ -44,6 +44,9 @@
 #endif
 #endif /* __linux__ */
 
+#if defined(__FreeBSD__) && !defined(HAVE_ENDIAN_H)
+#include <sys/endian.h>
+#endif
```
Comment (tobhe) [inline compat/endian.h:49]: I think we will still need the `#define betoh16 be16toh` part in this case.
So probably it should read:

```
#if defined(__FreeBSD__)
#if !defined(HAVE_ENDIAN_H)
#include <sys/endian.h>
#endif
#if !defined(betoh16)
#define betoh16	be16toh
...
```

Comment (DavidMarec) [reply compat/endian.h:50]: Yes, That is the lines following this statement that make the extra definitions

but your patch with the nested `defined`s works too

**Pull request merged**